### DISCUSSION TITLE
How to add new character classes?

### DISCUSSION THREAD

**Discussion by brockhaus** [Muttley/foundryvtt-shadowdark#630] [General]
So I would like to add new classes to my table. How to?

I started with the "Knight of St. Ydris" from Cursed Scroll 1, but how do I add "Class Talent Tables" (and what are they)? Are they hard coded or described by items, too?

**Comment by Muttley** [comment]
Just create a new "Class" Item and put it in a compendium (your own, not a system one unless you want to lose it after a system upgrade)

For the class talent table, just create a rollable table in a compendium and name it `Class Talents: <class_name>` where `<class_name>` is the actual class name that you're adding (must match exactly).  After that when you reopen the Class item the table will be available in the drop-down.

All custom background items you create must exist within a compendium before they are useable.

**Reply by brockhaus** [reply]
Hmm.. At the moment I try to add the class "Knight of St. Ydris". 

I have to add the talent "Demonic Possession". What do I add as effect to this talent?
When I look into "Backstab +1 Damage Dice" or "Weapon Mastery" they have an effect on "system.bonuses.backstabDie" or "system.bonuses.weaponMastery". Both are part of the code and have a translation.

So what do I call the effect for "Demonic Possession"? It would assume something like "system.bonuses.demonicPossession", but this is not part of the code obviously or translated.

Is ist possible even to add new class talents and with them classes w/o modifying code?
If so: Do you have a complete how to?
The class talent rolltables produce even more questions for me w/o documentation ..

**Reply by Muttley** [reply]
It's not quite as simple as that i'm afraid, we don't support all possible bonuses, especially things that aren't in the quickstart (our agreement with Arcane Library is that we can only include things from the Quickstart)

The best way would be to create a new Effect item, called Demonic Possession.  This can have the round duration set on it, and then add an active effect to it. Something like this:

![image](https://github.com/Muttley/foundryvtt-shadowdark/assets/853835/57eba47f-f94e-49f1-8a13-53636907b59b)

Then, when the player uses the ability, just drag the effect onto their sheet.  The rounds remaining will automatically be tracked and the player will have an effect tracking icon that shows on their screen showing how long is remaining.

If you have any more questions you'll probably get a quicker response on the official Arcane Library Discord.  There is a Foundry system thread on there where lots of people who use the system, as well as us developers, post:  https://discord.gg/thearcanelibrary

**Comment by brockhaus** [comment]
Awesome, thank you! I will try it like that. :)
